### PR TITLE
[Telink] Add Thermostat & Contact Sensor examples & Rework LED loop & made Factory-Reset user friendly

### PR DESCRIPTION
**Pull request by s07641069**
**Change overview**

- Added Telink Thermostat example (Cherrypick #23717)
- Used correct prototype of MatterPostAttributeChangeCallback for other Telink apps
- Cherrypick partly #23408 to make thermostat work with Google Home
- Added Telink Contact Sensor example
- Rework system status LED loop
- Made the Factory-Reset Button more user friendly

**Testing**

Tested manually with chip-tool and Google Home.
Steps:

Run: $ chip-tool pairing ble-thread <...>
Wait till success

### CODE REVIEW

**Comment by s07641069** [comment]
@andy31415 I updated Telink Docker file just as reminder for developers which Zephyr HASH is compatible.
Docker image rebuild not needed.

File: integrations/docker/images/chip-build-telink/Dockerfile [hunk 1]
```diff
@@ -17,7 +17,7 @@ RUN set -x \
     && : # last line
 
 # Setup Zephyr
-ARG ZEPHYR_REVISION=295b4e6dedc4f40701962e0ed0be1090be5c0eee
+ARG ZEPHYR_REVISION=dac17979d626bb820268660b4272a56d04d8083b
```
Comment (andy31415) [inline integrations/docker/images/chip-build-telink/Dockerfile:20]: This does not work: we need dockerfiles to be a fixed version that is always identican when built.

If we change this, we will make a telink docker that may have different content depending on what version was built.

If you want, you could delete the docker file from here and add a README pointing to what docker file needs to be used. Or use some different numbering scheme (but then docker rebuilds will be needed).

Unfortunately we never had this situation (a branch needing a different docker) so we do not have a clear way to handle it. Maybe just changing version in `.github/workflows` also works.

File: .github/workflows/examples-telink.yaml [hunk 2]
```diff
@@ -99,6 +99,15 @@ jobs:
                     out/telink-tlsr9518adk80d-ota-requestor/zephyr/zephyr.elf \
                     /tmp/bloat_reports/
 
+            - name: Build example Telink Thermostat App
```
Comment (andy31415) [inline .github/workflows/examples-telink.yaml:102]: did the dockerfile version not need changing in this file?

File: integrations/docker/images/chip-build-telink/Dockerfile (hunk 1 above)
Comment (s07641069) [reply integrations/docker/images/chip-build-telink/Dockerfile:20]: I mean that in .github/workflows/examples-telink.yaml we already changed telink docker image version for 1.0-branch previously. 
And connectedhomeip/chip-build-telink:0.6.06 image which is active for Telibk 1.0-branch is builded from Zephyr hash dac17979d626bb820268660b4272a56d04d8083b.
Zephyr hash 295b4e6dedc4f40701962e0ed0be1090be5c0eee - is pretty old one and compatible with Telink commit under release tag, but not with latest commit in 1.0-branch.
I make misunderstanding for our customers when they use 1.0-branch.
But its you choose, it this is not allowed, I will revert this change .

File: .github/workflows/examples-telink.yaml (hunk 2 above)
Comment (s07641069) [reply .github/workflows/examples-telink.yaml:102]: No

File: integrations/docker/images/chip-build-telink/Dockerfile (hunk 1 above)
Comment (s07641069) [reply integrations/docker/images/chip-build-telink/Dockerfile:20]: ok, Have I revert this change in docker file? This change is not mandatory for us, just as pointer of hash into corresponding docker which used in actual commit.

Comment (andy31415) [reply integrations/docker/images/chip-build-telink/Dockerfile:20]: Yes, please revert this change. The rest should be good to go.

Comment (s07641069) [reply integrations/docker/images/chip-build-telink/Dockerfile]: Done

**Comment by s07641069** [comment]
@andy31415 Could you review and process this cherry-pick PR?

**Review by andy31415 — APPROVED** [review]

**Pull request merged**